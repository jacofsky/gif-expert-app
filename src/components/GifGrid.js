import React, {} from 'react'
import PropTypes from 'prop-types'

import GifGridItem from './GifGridItem'
import { useFetchGifs } from '../hooks/useFetchGifs'

const GifGrid = ({category}) => {

    const {data:images, loading} = useFetchGifs(category);

    return (
        <div>
            <h3>{category}</h3>
            {loading && 'Cargando...'}

            <ul className='cardList'>
                {images.map(image => <GifGridItem key={image.id} {... image}/> )}
            </ul>
            
        </div>
    )
}

GifGrid.prototype = {
    category: PropTypes.string.isRequired
}

export default GifGrid
