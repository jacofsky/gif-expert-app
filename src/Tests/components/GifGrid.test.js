import react from "react";
import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs');

describe('Tests in GifGrid.test.js', () => {

    const category = 'One Punch'
    
    test('should match the snapshot', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })

        const wrapper = shallow(<GifGrid category = {category}/>)
        expect(wrapper).toMatchSnapshot();
    })
    
    test('should show items when img have been upload', () => {
        // deberemos hacer un mock para hacer de cuenta de que el custom hook ya recibio la informacion
        
        const gifs = [{
            id: 'ABC',
            img: 'https://localhost/cualquiercosa.jpg',
            title: 'AAA',
            username: 'AAAS'
        },{
            id: '123',
            img: 'https://localhost/cualquiercosa.jpg',
            title: 'AAA',
            username: 'AAAS'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })

        const wrapper = shallow(<GifGrid category = {category}/>)
        
        expect(wrapper.find('p').exists()).toBe(false)
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
    })
    
})
