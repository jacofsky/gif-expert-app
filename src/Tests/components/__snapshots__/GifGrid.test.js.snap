// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in GifGrid.test.js should match the snapshot 1`] = `
<div>
  <h3>
    One Punch
  </h3>
  Cargando...
  <ul
    className="cardList"
  />
</div>
`;
